### DISCUSSION TITLE
i want to startup few applications when i open i3

### DISCUSSION THREAD

**Discussion by rustedusted** [i3/i3#6302] [Q&A]
```
exec --no-startup-id sh -c 'i3-msg "workspace 1;exec /usr/bin/alacritty"'
exec --no-startup-id sh -c 'i3-msg "workspace 2;exec /usr/bin/firefox --kiosk"'
exec --no-startup-id sh -c 'i3-msg "workspace 5;exec /usr/bin/firefox -p SolderedHuman --new-window https://music.youtube.com/"'
exec --no-startup-id sh -c 'i3-msg "workspace 3;exec /usr/bin/gtk-launch ferdium"'


```
this is what i've wrote
but this opens things in wrong workspaces i don't know why tbh

**Comment by okraits** [comment]
I recommend to use the `assign` directive to assign applications to specific workspaces:

https://i3wm.org/docs/userguide.html#assign_workspace

**Reply by rustedusted** [reply]
as i've understood assign makes it such that all windows of that class open in that exact workspace 

is there a way to unassign after the windows have opened then

so that i can assign and after executing the commmand unassign them
so that i am able to open these windows in other workspaces

**Reply by rustedusted** [reply]
i want firefox to open on workspace 2 when i start computer and if i open firefox any other way after startup that would open on the workspaced that i'm focused on

**Comment by baruch** [comment]
The commands are all running in parallel, they switch the workspace in random order and run the programs but they do not wait for the program to open before switching to another workspace.

That can explain why it happens but I don't have a solution for you. Sorry.

**Reply by rustedusted** [reply]
oh well that's unfortunate well i wish there was a way to give window opening commands a variable saying this is the currently focused window and not what the actual variables

this could emulate the behaviour of opening things in different workspaces

**Reply by orestisfl** [reply]
Executables and windows have a many-to-many relation. An executable can open zero, one or more windows. Assigning windows to the workspace they were initiated from is part of the [startup-notification-spec](https://www.freedesktop.org/wiki/Specifications/startup-notification-spec/) that i3 does support.
